### PR TITLE
Add flag to draw MCParticles that were created in the simulation

### PR DESCRIPTION
**Pull request by Zehvogel**
BEGINRELEASENOTES
- Added the processor parameter `DrawMCParticlesCreatedInSimulation` to enable drawing of the MCParticles that were created in the simulation.

ENDRELEASENOTES

solves #26

### CODE REVIEW

File: src/DDCEDViewer.cc [hunk 1]
```diff
@@ -649,19 +654,22 @@ void DDCEDViewer::drawMCParticle(dd4hep::Detector& theDetector, int& layer, unsi
     for(int i=0; i<col->getNumberOfElements() ; i++){
         MCParticle* mcp = dynamic_cast<MCParticle*> ( col->getElementAt( i ) ) ;
         float charge = mcp->getCharge ();
-        if( mcp-> getGeneratorStatus() != 1 && _usingParticleGun == false ) 
-            continue ; // stable particles only
-        if ( mcp->getEnergy() < _mcpECut ) 
+        if( mcp-> getGeneratorStatus() != 1 && _usingParticleGun == false ) {
+            if (! (_drawMCParticlesCreatedInSimulation && mcp->isCreatedInSimulation() )) {
```
[the remark this answers is not in the document]
Comment (Zehvogel) [reply src/DDCEDViewer.cc:658]: do you have a suggestion?

Comment (tmadlener) [inline src/DDCEDViewer.cc:658]:
```suggestion
        if( mcp-> getGeneratorStatus() != 1 && 
             _usingParticleGun == false 
             ! (_drawMCParticlesCreatedInSimulation && mcp->isCreatedInSimulation() ) {
```
(plus removing additional braces below). Unless, I am missing something here, this should work, no?

**Comment by Zehvogel** [comment]
any other change wishes? :)

**Comment by tmadlener** [comment]
I just realize we have no working CI any longer. I think this looks good, but I would like to fix CI to make sure we don't break anything unexpectedly

**Review by tmadlener — APPROVED** [review]

**Pull request merged**